### ISSUE TITLE
Unchanged values unexpectedly marked for update

### ISSUE THREAD

**Issue by lwansbrough** [balderdashy/sails#5459]
Original issue here: https://github.com/balderdashy/sails/issues/2297

The TL;DR of that issue is that the Sails docs say that when `beforeUpdate` is called, the `valuesToUpdate` argument will contain the values that are being updated (this is what I've interpreted it to mean, anyway). This makes sense and should be the way it works. However, this is only true for updates made through the static `Model.update(...)` method, and not the `model.save()` instance method. In the latter case, `valuesToUpdate` will be all the properties of the model instance, not just the values being updated. Typically the suggestion could just be to use the static method if that's a problem, but this is not possible if you need to update an association. Updating an association currently requires using the model instance.

The solution to this problem therefore, is that we either need static methods for adding associations, or the implementation for updating model instances needs to be changed. I prefer both be done actually, I think there's valid reasons to use both. I've outlined how this might be accomplished in the other issue.

Thanks for your consideration guys.
